libs/shared/ui/src/lib/components/inputs/input-select/input-select.tsx

```diff
@@ -13,20 +13,38 @@ import Select, {
   components,
 } from 'react-select'
 import CreatableSelect from 'react-select/creatable'
-import { match } from 'ts-pattern'
-import { type Value } from '@qovery/shared/interfaces'
+import { P, match } from 'ts-pattern'
+import { type SelectOption, type SelectOptionValue } from '@qovery/shared/interfaces'
 import { Icon } from '../../icon/icon'
 import { LoaderSpinner } from '../../loader-spinner/loader-spinner'
 
+// Helper function to compare SelectOptionValue types
+const isEqualValue = (a: SelectOptionValue, b: SelectOptionValue): boolean => {
+  if (typeof a === 'string' && typeof b === 'string') {
+    return a === b
+  }
+
+  if (typeof a === 'object' && typeof b === 'object' && a !== null && b !== null) {
+    // For GitRepository objects, compare by name property
```

Comment:
(comment) I'm not sure it's a good thing to have a Qovery technical logic inside design system component no?

Reply:
Very good remark! I'll tackle this in another PR